overlord/snapstate/snapstate.go

```diff
@@ -266,13 +267,27 @@ var snapTopicalTasks = map[string]bool{
 	"unalias":            true,
 	"disable-aliases":    true,
 	"prefer-aliases":     true,
+	"connect":            true,
+	"disconnect":         true,
+}
+
+func getPlugAndSlotRefs(task *state.Task) (interfaces.PlugRef, interfaces.SlotRef, error) {
+	var plugRef interfaces.PlugRef
+	var slotRef interfaces.SlotRef
+	if err := task.Get("plug", &plugRef); err != nil {
+		return plugRef, slotRef, err
+	}
+	if err := task.Get("slot", &slotRef); err != nil {
+		return plugRef, slotRef, err
```

Comment:
this is returning structs, not pointer to structs, maybe that should indeed be changed though